packages/server/graphql/mutations/helpers/generateGroups.ts

```diff
@@ -17,18 +17,24 @@ const generateGroups = async (
   if (!hasSuggestGroupsFlag) return
   const groupReflectionsInput = reflections.map((reflection) => reflection.plaintextContent)
   const manager = new OpenAIServerManager()
-  const groupedReflectionsJSON = await manager.groupReflections(groupReflectionsInput)
-  if (!groupedReflectionsJSON) {
+
+  const themes = await manager.generateThemes(groupReflectionsInput)
+  if (!themes) {
+    console.warn('ChatGPT was unable to generate themes')
+    return
+  }
+  const groupedReflections = await manager.groupReflections(groupReflectionsInput, themes)
+
+  if (!groupedReflections) {
     console.warn('ChatGPT was unable to group the reflections')
     return
   }
-  const parsedGroupedReflections = JSON.parse(groupedReflectionsJSON)
   const autogroupReflectionGroups: AutogroupReflectionGroupType[] = []
 
-  for (const group of parsedGroupedReflections) {
-    const groupTitle = Object.keys(group)[0]
+  for (const groupTitle of Object.keys(groupedReflections)) {
```

Comment:
Alternative approach `for (const [groupTitle, reflectionTexts] of Object.entries(groupedReflections)) {`

Reply:
Oh much cleaner, updated!